joblib/test/test_hashing.py

```diff
@@ -70,8 +72,8 @@ def f(self, x):
 
 class KlassWithCachedMethod(object):
 
-    def __init__(self):
-        mem = Memory(cachedir=test_memory_env['dir'])
+    def __init__(self, tmpdir_path):
```

Comment:
Just curious, do you need `tmpdir_path` in the `__init__` arguments, i.e. could you not just have `mem = Memory(cachedir=tmpdir_path)`?